src/ast/dibuilderbpf.cpp

```diff
@@ -227,8 +227,8 @@ DIType *DIBuilderBPF::GetMapKeyType(const MapKey &key,
     extra_arg_size = 8;
 
   // Single map key -> use the appropriate type.
-  if (key.args_.size() == 1 && extra_arg_size == 0)
-    return GetType(key.args_[0]);
+  if (extra_arg_size == 0)
+    return GetType(key.arg_);
 
   // Multi map key -> use byte array.
```

Comment:
I need to read the rest of this PR more carefully, but "multi map key" logic can be deleted. After this change, all maps will have 0 or 1 keys. My hope was that it'll get rid of a lot of redundant code.

Reply:
This alone would be a good reason for going forward with this :-)

Reply:
Maybe. Right now we still have multi-map keys for `hist` and `lhist`. I can look at refactoring that but it might be adding more scope to this change.